src/adapter/localFolder.ts

```diff
@@ -22,36 +22,38 @@ class LocalFolderAdapter extends BaseAdapter {
         });
     }
 
-    requestRandomImage(): Promise<HistoryEntry> {
+    requestRandomImage(count: number): Promise<HistoryEntry[]> {
         return new Promise((resolve, reject) => {
             const folder = Gio.File.new_for_path(this._settings.getString('folder'));
             const files = this._listDirectory(folder);
+            const wallpaperResult: HistoryEntry[] = [];
 
             if (files.length < 1) {
                 reject(new Error('No files found'));
                 return;
             }
 
-            let randomFilePath: string | null = null;
-            for (let i = 0; i < 5; i++) {
+            for (let i = 0; i < 20 && wallpaperResult.length < count; i++) {
                 const randomFile = files[Utils.getRandomNumber(files.length)];
-                randomFilePath = randomFile.get_uri();
+                const randomFilePath = randomFile.get_uri();
                 const randomFileName = randomFile.get_basename();
 
-                if (randomFileName && !this._isImageBlocked(randomFileName))
-                    break;
+                if (!randomFileName || this._isImageBlocked(randomFileName))
+                    continue;
 
-                randomFilePath = null;
+                const historyEntry = new HistoryEntry(null, this._sourceName, randomFilePath);
+                historyEntry.source.sourceUrl = this._wallpaperLocation;
+
+                if (!this._includesWallpaper(wallpaperResult, historyEntry.source.imageDownloadUrl))
+                    wallpaperResult.push(historyEntry);
             }
 
-            if (!randomFilePath) {
+            if (wallpaperResult.length === 0) {
```

Comment:
> Should check for < count instead.

Currently, it only fails if no new images will be found (=== 0) this will escalate back upwards and results in no new images being set. If at least one new image is found it won't fail and fill missing images randomly from the history later on.
I can see that this will still fail when only one of multiple adapters fail. I should try to count the failing promises here.

But from your comment I assume you want to fail already if not all requested images can be fulfilled?

> Maybe also log a warning and return some partial results (this is just an idea).

Added log warning.
Partial results are already happening, see above.

Reply:
> But from your comment I assume you want to fail already if not all requested images can be fulfilled?
Basically yes, I would have expected it to fail just based on the function signature "request N images".

I think I'll have to take some more time to get more familiar with all changes and experiment with the code a bit.

Reply:
Sure, I'll change it to this behavior. :+1:

Reply:
[`44a6a76` (#154)](https://github.com/ifl0w/RandomWallpaperGnome3/pull/154/commits/44a6a76364b422e9b39d2119abe8dc964ddff73f)